packages/patternfly-3/patternfly-react/src/components/DateTimePicker/DateComponents/Header.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Icon } from '../../../index';
+import { addDays } from './helpers';
+import times from 'lodash/times';
+
+class Header extends React.Component {
+  getWeekStart = date => addDays(date, (7 - date.getDay()) % 7);
+  weekArray = () => {
```

Comment:
as suggested on other places, maybe move this logic into helpers and keep this component simple.